### PR TITLE
Fixed tp implementation for resistive mikromedia boards

### PR DESCRIPTION
**Pull request by IvanRuzavin**
This PR is connected to #[SWMIKSDK-1697](https://jira.mikroe.com/browse/SWMIKSDK-1697)

Changes:

- [middleware/tsc2003/lib/src/tsc2003.c](https://github.com/MikroElektronika/mikrosdk_v2/blob/fix/touch_panel_resistive_tsc2003/middleware/tsc2003/lib/src/tsc2003.c)
    - removed pen pressure detection as it doesn't work with TSC2003
- [api/tp/lib/src/tp/tp.c](https://github.com/MikroElektronika/mikrosdk_v2/blob/fix/touch_panel_resistive_tsc2003/api/tp/lib/src/tp/tp.c)
    - added option to clear all the touches for TSC2003 specifically

Fixes:

- now touch is detected and cleared while using TSC2003 display with LVGL project
NOTE: STMPE approach for clearing touch inside the library itself doesn't work as it differs because it uses SPI - it sends the GPIO touch detection signal and then clears it even if we didn't release the pen whereas TSC is always sending GPIO touch detection signal until we stop touching the screen, so clearing of the touch should be done in tp.c.

### CODE REVIEW

File: changelog/v2.11.3/changelog.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+---
+
+**[BACK TO MAIN FILE](../../changelog.md)**
+
+---
+
+# `v2.11.3`
+
++ released: TBD
```
Comment (StrahinjaJacimovic) [inline changelog/v2.11.3/changelog.md:13]: Release is planned for October 1-st this year. Update the year accordingly here.

File: api/tp/lib/src/tp/tp.c [hunk 2]
```diff
@@ -511,6 +511,10 @@ tp_process( tp_t * ctx )
         ctx->gesture_prev         = TP_EVENT_GEST_NONE;
         ctx->release              = TP_RELEASE_DET;
         ctx->touch_prev.n_touches = 0;
+        // For TSC2003 display there can be only 1 touch, so if display is not pressed, there are no touches.
+        #ifdef __TP_TSC2003__
+        ctx->touch.n_touches = 0;
```
Comment (StrahinjaJacimovic) [inline api/tp/lib/src/tp/tp.c:516]: How does this affect capacitive displays? If at all.

**Review by StrahinjaJacimovic — CHANGES_REQUESTED** [review]

File: api/tp/lib/src/tp/tp.c (hunk 2 above)
Comment (IvanRuzavin) [reply api/tp/lib/src/tp/tp.c:516]: It affects only old Resistive Displays - their touch detection approach is different from TSC2003 or FT5XX6 as STMPE protocol clears the touch detections on HW level and we handled this detection via SW. So if we clear the touch detection info here, the touch doesn't work as expected (for example, you can't hold the button pressed).

As for Capacitive displays - you are correct, by the time we get here, this variable value is already 0 as we release all the pens from the Display.

File: changelog/v2.11.3/changelog.md (hunk 1 above)
Comment (IvanRuzavin) [reply changelog/v2.11.3/changelog.md]: Updated date accordingly.

**Review by StrahinjaJacimovic — APPROVED** [review]

**Pull request merged**